src/drivers/actuators/vertiq_io/vertiq_io.hpp

```diff
@@ -0,0 +1,200 @@
+#include <uORB/topics/esc_status.h>
+#include <uORB/topics/actuator_test.h>
+
+#include "vertiq_telemetry_manager.hpp"
+#include "vertiq_client_manager.hpp"
+#include "vertiq_serial_interface.hpp"
+
+#include "iq-module-communication-cpp/inc/propeller_motor_control_client.hpp"
+#include "iq-module-communication-cpp/inc/brushless_drive_client.hpp"
+#include "iq-module-communication-cpp/inc/arming_handler_client.hpp"
+
+class VertiqIo : public ModuleBase<VertiqIo>, public OutputModuleInterface
+{
+
+public:
+
+	/**
+	* @brief Create a new VertiqIo object
+	*/
+	VertiqIo();
+
+	/**
+	* @brief destruct a VertiqIo object
+	*/
+	~VertiqIo() override;
+
+	/**
+	* @brief initialize the VertiqIo object. This will be called by the task_spawn function. Makes sure that the thread gets scheduled.
+	*/
+	bool init();
+
+	/** @see ModuleBase */
+	static int task_spawn(int argc, char *argv[]);
+
+	/** @see ModuleBase */
+	static int custom_command(int argc, char *argv[]);
+
+	/** @see ModuleBase */
+	static int print_usage(const char *reason = nullptr);
+
+	int print_status() override;
+
+	/** @see ModuleBase::run() */ //I do not think this actually comes from ModuleBase. it should come from scheduled work item
+	void Run() override;
+
+	/** @see OutputModuleInterface */
+	bool updateOutputs(bool stop_motors, uint16_t outputs[MAX_ACTUATORS],
+			   unsigned num_outputs, unsigned num_control_groups_updated) override;
+
+	/**
+	* @brief Used to package and transmit controls via IQUART
+	* @param outputs The output throttles calculated by the mixer
+	*/
+	void OutputControls(uint16_t outputs[MAX_ACTUATORS]);
+
+private:
+	static const uint8_t MAX_SUPPORTABLE_IFCI_CVS = 16;
+
+	enum DISARM_BEHAVIORS {TRIGGER_MOTOR_DISARM, COAST_MOTOR, SEND_PREDEFINED_THROTTLE};
+	enum ARM_BEHAVIORS {USE_MOTOR_ARMING, FORCE_ARMING};
+
+	//Variables and functions necessary for properly configuring the serial interface
+	//Determines whether or not we should initialize or re-initialize the serial connection
+	static px4::atomic_bool _request_telemetry_init;
+
+	MixingOutput _mixing_output{"VERTIQ_IO", MAX_SUPPORTABLE_IFCI_CVS, *this, MixingOutput::SchedulingPolicy::Auto, false, false};
+
+	//The name of the device we're connecting to. this will be something like /dev/ttyS3
+	static char _telemetry_device[20];
+
+	//Counters/timers to track our status
+	perf_counter_t	_loop_perf{perf_alloc(PC_ELAPSED, MODULE_NAME": cycle")};
+	perf_counter_t	_loop_interval_perf{perf_alloc(PC_INTERVAL, MODULE_NAME": output update interval")};
+
+	//We need a serial handler in order to talk over the serial port
+	VertiqSerialInterface _serial_interface;
+
+	//We need someone who can manage our clients
+	VertiqClientManager _client_manager;
+
+	//We need a telemetry handler
+	VertiqTelemetryManager _telem_manager;
+
+	static const uint16_t MAX_IFCI_MESSAGE = 40; //Up to 16 2 byte commands, one telemetry byte, plus 7 IQUART added bytes
+
+	IQUartFlightControllerInterfaceClient _operational_ifci;
```

Comment:
Another place where commenting would be good to explain the need for the operational IFCI versus the other IFCIs kicking around. Considering how many of these IFCIs there are, a more descriptive name than operational may got a good bit of the way to clearing up the specific purpose of this one. One could argue the "telem" one is just as operational